### DISCUSSION TITLE
Получение DelaySend документов через DocumentsTask

### DISCUSSION THREAD

**Discussion by bBubbleGum** [diadoc/diadocsdk-1c-docs#891] [General]
Здравствуйте. С помощью компоненты AddIn  получаю все исходящие документы с параметрами 
FromSendDate и ToSendDate. 
Но за указанный интервал выгружаются не все документы. Почему такое возможно?

```
SearchDocumentTask = Organization.GetDocumentsTask();	
НачалоПериода = НачалоДня('20220826');
КонецПериода = КонецДня('20220826');
	
SearchDocumentTask.FromSendDate = НачалоПериода;
SearchDocumentTask.ToSendDate   = КонецПериода;
	
 //Исходящие
SearchDocumentTask.Category = "Any.Outbound"; 
	
DocumentList = SearchDocumentTask.GetDocuments();
```

**Comment by GilimkhanovDenis** [comment]
Добрый день

нужен пример документа, который не попал в выборку

**Comment by bBubbleGum** [comment]
> Добрый день
> 
> нужен пример документа, который не попал в выборку

А куда мне прислать Ид документа?

**Comment by GilimkhanovDenis** [comment]
Можете здесь его написать

нужна ссылка из веб или DocumentBase.PathUrl из компоненты

**Comment by bBubbleGum** [comment]
> Можете здесь его написать
> 
> нужна ссылка из веб или DocumentBase.PathUrl из компоненты

У документа статус "Требуется подписать и отправить", ДопСтатус "На подписании". 
И как я поняла, у него нет метки даты, что он отправлен, и поэтому он не попадает в выборку 
с параметрами `FromSendDate и ToSendDate`. Но в модуле Диадок этот документ отражается, если задан отбор по дате отправления. Как тогда нужно получать такие документы, если у компоненты нет других параметров, кроме как 
`FromSendDate ,  ToSendDate` и `FromDocumentDate, ToDocumentDate ` ?
Ссылка на документ:
`/aa8fa5a5-33b7-4bf4-a549-ce7f42737049/Document/Show?letterId=6f494b5b-a465-4b53-8e39-56fc2b613174&documentId=6b19c79e-1044-4b30-a257-52544ab47fa1`

**Comment by GilimkhanovDenis** [comment]
Попробуйте убрать верхнюю границу (КонецПериода).

Считается, что у документов с отложенной отправкой дата отправки далеко в будущем

**Comment by bBubbleGum** [comment]
> Попробуйте убрать верхнюю границу (КонецПериода).
> 
> Считается, что у документов с отложенной отправкой дата отправки далеко в будущем

Тогда не получится ограничить отбор за период. Если я хочу получить документы только за август, в выгрузке будут и документы за сентябрь

**Comment by bBubbleGum** [comment]
Плохо, что в свойстве `ResolutionStatus `объекта `Document `нет параметра даты создания. Для получения даты создания документа придется использовать метод `GetResolutions()`. Без верхней границы интервала с учетом всех проверок на дату как-то не оптимально получается...
![image](https://user-images.githubusercontent.com/113088500/189294298-57cb4f3c-fb5e-45dc-af0c-ceb7cb74847d.png)

**Comment by GilimkhanovDenis** [comment]
Без доработок на стороне сервиса Диадок не сможем добавить получение ещё не отправденных документов в компоненту

Если функциональность крайне необходима, то можете обратиться к вашему менеджеру и через него продвигать задачу в очереди разработки

**Reply by bBubbleGum** [reply]
спасибо за ответ. Но я уже решила задачу с помощью того, что есть)
